bitcoin/src/blockdata/transaction.rs

```diff
@@ -1192,7 +1186,10 @@ pub fn effective_value(
     satisfaction_weight: Weight,
     value: Amount,
 ) -> Option<SignedAmount> {
-    let weight = satisfaction_weight.checked_add(TxIn::BASE_WEIGHT)?;
+    let previous_output_weight = Weight::from_vb_unwrap(OutPoint::SIZE as u64);
+    let sequence_weight = Weight::from_vb_unwrap(Sequence::SIZE as u64);
+    let weight = satisfaction_weight.checked_add(previous_output_weight)?.checked_add(sequence_weight)?;
```

Comment:
Just realized, the original was more efficient because it had one-less overflow check. Maybe the compiler can remove it but I wouldn't bet on it here.

I suggest you add the constants together without check.